drivers/can/can_stm32.c

```diff
@@ -1042,6 +1087,9 @@ static const struct can_driver_api can_api_funcs = {
 
 static void config_can_1_irq(CAN_TypeDef *can);
 
+static const struct soc_gpio_pinctrl pins_can_1[] =
+	ST_STM32_DT_PINCTRL(can1, 0);
```

Comment:
@erwango This even compiles if I use random arguments like `ST_STM32_DT_PINCTRL(foo, bar);`. I will also test it in hardware to verify it's correct, but I think it would be nice to get a compile error if something non-existing is passed.

Reply:
This is strange, this stuff is usually generous in compilation errors.
I'll have a check

Reply:
ok, got it.
Macro is built as follows:
```
	{ COND_CODE_1(DT_NODE_HAS_PROP(DT_NODELABEL(name), pinctrl_##x),\
		      (UTIL_LISTIFY(ST_STM32_DT_NUM_PINS(name, x),	\
				   ST_STM32_DT_PIN_ELEM,		\
				   x,					\
				   name)),				\
		      ())						\
	}
```
So indeed, if `DT_NODE_HAS_PROP(DT_NODELABEL(name), pinctrl_##x)` returns `false`, which it should do for `(foo, bar)`, nothing will happen.

We can remove the use of `COND_CODE_1`, but then it will report an error for each node that doesn't have a pinctrl-x property. We'll be able to do that, but later, once the pinmux.c files have been removed (after DV2.6 ?).